### PR TITLE
Fill out shared::ntdef

### PR DESCRIPTION
**Pull request by docbrown**
<del>Microsoft uses a tool to build winnt.h from ntdef.h, translating NT types to their Win32 equivalents (USHORT becomes WORD, ULONG becomes DWORD, and so on). Since the NT and Win32 types use the same underlying Rust types, we can just re-export a lot of ntdef items in winnt without issue.</del>

<del>I've also fixed up some of the cfg attributes so they are closer to what's in the headers and implemented some inline functions.</del>

I've reduced the scope of this PR to just include changes to ntdef.

### CODE REVIEW

**Comment by retep998** [comment]
PR has conflicts.

**Comment by retep998** [comment]
When it comes to duplicated definitions between headers, I settled on this as a policy: https://github.com/retep998/winapi-rs/blob/dev/CONTRIBUTING.md#dealing-with-duplicates
Could you update the winnt stuff to follow that?

**Comment by Susurrus** [comment]
@docbrown Any interest in fixing the issues with this PR so it can be merged?

**Comment by docbrown** [comment]
@Susurrus I lost my Windows dev VM, but I'm in the process of getting everything installed on my desktop. I'll come back to this shortly.

**Comment by Susurrus** [comment]
@docbrown you can actually do all the development on Linux and use WINE for testing. It's what I do. You'll just need a zip of the header docs for a reference which I got from @retep998 and can send you the link for.

**Comment by docbrown** [comment]
@Susurrus Thanks, but I got all of my tools set up again. I've added a few things to `ntdef` from the latest SDK and am currently cleaning up the rest of `winnt`. It's a comically large header, though, so it's taking longer than I expected.

**Comment by Susurrus** [comment]
Sounds good! Depending on what files you're touching you might want to base your branch off #447 as he does some changes to `winnt`. They may overlap with what you're doing.

**Comment by retep998** [comment]
Or you could simply base your branch on dev and rebase as needed.

**Comment by docbrown** [comment]
`winnt` is a beast, so I'm just going to make those changes in a separate PR, perhaps after #447 is merged. This PR should be complete now.

File: src/shared/ntdef.rs [hunk 1]
```diff
@@ -176,8 +197,872 @@ pub const MAXLONGLONG: LONGLONG = 0x7fffffffffffffff;
 pub type PLONGLONG = *mut LONGLONG;
 pub type PULONGLONG = *mut ULONGLONG;
 pub type USN = LONGLONG;
-pub type LARGE_INTEGER = LONGLONG;
+UNION2!{union LARGE_INTEGER {
+    [LONGLONG; 1],
```
Comment (retep998) [inline src/shared/ntdef.rs]: The storage for unions should be the primitive integer types like `u64`.

File: src/shared/ntdef.rs [hunk 2]
```diff
@@ -176,8 +197,872 @@ pub const MAXLONGLONG: LONGLONG = 0x7fffffffffffffff;
...
+STRUCT!{struct RTL_BALANCED_NODE_s {
+    Left: *mut RTL_BALANCED_NODE,
+    Right: *mut RTL_BALANCED_NODE,
+}}
+pub const RTL_BALANCED_NODE_RESERVED_PARENT_MASK: ULONG_PTR = 3;
+pub type PRTL_BALANCED_NODE = *mut RTL_BALANCED_NODE;
+#[inline]
+pub unsafe fn RTL_BALANCED_NODE_GET_PARENT_POINTER(Node: PRTL_BALANCED_NODE)
+    -> PRTL_BALANCED_NODE
+{
```
Comment (retep998) [inline src/shared/ntdef.rs]: Block indentation:
```Rust
pub unsafe fn RTL_BALANCED_NODE_GET_PARENT_POINTER(
    Node: PRTL_BALANCED_NODE,
) -> PRTL_BALANCED_NODE {
```

File: src/shared/ntdef.rs [hunk 3]
```diff
@@ -176,8 +197,872 @@ pub const MAXLONGLONG: LONGLONG = 0x7fffffffffffffff;
...
+pub const SORT_KOREAN_KSC: USHORT = 0x0;
+pub const SORT_KOREAN_UNICODE: USHORT = 0x1;
+pub const SORT_GERMAN_PHONE_BOOK: USHORT = 0x1;
+pub const SORT_HUNGARIAN_DEFAULT: USHORT = 0x0;
+pub const SORT_HUNGARIAN_TECHNICAL: USHORT = 0x1;
+pub const SORT_GEORGIAN_TRADITIONAL: USHORT = 0x0;
+pub const SORT_GEORGIAN_MODERN: USHORT = 0x1;
+macro_rules! MAKELANGID { ($p:expr, $s:expr) => ((($s as USHORT) << 10) | ($p as USHORT)) }
+#[inline]
+pub fn MAKELANGID(p: USHORT, s: USHORT) -> LANGID { (s << 10) | p }
```
Comment (Susurrus) [inline src/shared/ntdef.rs]: Should these fns all be `unsafe` as well? Additionally, I think they need block indentation for their arguments.

File: src/shared/ntdef.rs [hunk 4]
```diff
@@ -5,22 +5,27 @@
 // All files in the project carrying such notice may not be copied, modified, or distributed
 // except according to those terms.
 //! Type definitions for the basic types.
+use core::{mem, ptr};
```
Comment (retep998) [inline src/shared/ntdef.rs]: Would it be possible to use the specific items instead of the modules?

Comment (docbrown) [inline src/shared/ntdef.rs]: Importing the items individually breaks the check_imports test.

**Comment by retep998** [comment]
Why did you make those functions unsafe? If all they do is safe things, then they can remain safe.

File: src/shared/ntdef.rs (hunk 4 above)
Comment (retep998) [inline src/shared/ntdef.rs]: What if you imported these items directly inside the function where you used them?

**Pull request merged**